labrad/wrappers.py

```diff
@@ -390,18 +390,76 @@ def _send(self, *args, **kw):
         return self._cxn._send(self.ID, *args, **kw)
 
 @inlineCallbacks
-def getConnection(host=C.MANAGER_HOST, port=C.MANAGER_PORT, name="Python Client", password=None):
+def getConnection(host=C.MANAGER_HOST, port=None, name="Python Client", password=None, tls=C.MANAGER_TLS):
     """Connect to LabRAD and return a deferred that fires the protocol object."""
-    p = yield protocol.factory.connectTCP(host, port, C.TIMEOUT)
+    if port is None:
+        port = C.MANAGER_PORT_TLS if tls.lower() == 'on' else C.MANAGER_PORT
```

Comment:
the tls mode does have three possible values: 'on', 'off' and 'starttls'. I've added a function `labrad.constants.check_tls_mode` to validate that this parameter is one of the allowed options and to coerce it to canonical form (lower case).